test/suite/test_backup_base.py

```diff
@@ -0,0 +1,77 @@
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+import os
+import wttest
+from suite_subprocess import suite_subprocess
+from helper import compare_files
+
+# Shared base class used by backup tests.
+class test_backup_base(wttest.WiredTigerTestCase, suite_subprocess):
+    #
+    # Set up all the directories needed for the test. We have a full backup directory for each
+    # iteration and an incremental backup for each iteration. That way we can compare the full and
+    # incremental each time through.
+    #
+    def setup_directories(self, max_iteration, home_incr, home_full, logpath):
+        for i in range(0, max_iteration):
+            # The log directory is a subdirectory of the home directory,
+            # creating that will make the home directory also.
+
+            home_incr_dir = home_incr + '.' + str(i)
+            if os.path.exists(home_incr_dir):
+                os.remove(home_incr_dir)
+            os.makedirs(home_incr_dir + '/' + logpath)
+
+            if i == 0:
+                continue
+            home_full_dir = home_full + '.' + str(i)
+            if os.path.exists(home_full_dir):
+                os.remove(home_full_dir)
+            os.makedirs(home_full_dir + '/' + logpath)
+    #
+    # Compare against two directory paths using the wt dump command. If the second directory path is not provided
+    # it checks if the compare
+    #
+    def compare_backups(self, uri, dir1, file1, run_recovery = False, dir2 = None, file2 = None):
+        if os.path.exists(file1):
+            os.remove(file1)
+        # Run wt dump on first backup directory.
+        self.runWt(['-R', '-h', dir1, 'dump', uri], outfilename=file1)
+        # Run wt dump on second backup directory if provided.
+        # If not provided, run dump on original directory, and check if we run recovery or not
+        if dir2 == None:
+            file2 = 'orig'
+            if os.path.exists(file2):
+                os.remove(file2)
+            if run_recovery:
+                self.runWt(['-R', 'dump', uri], outfilename=file2)
```

Comment:
Hey @sueloverso , wanted your insight on this, can we force recovery with **-R** with tests that are don't test the incremental backup? If we can, we wouldn't need the **run_recovery**

Reply:
This appears to be resolved.

Reply:
Hey @sueloverso ! actually we still need your insight in this, we are still not 100% sure that we should forcing recovery on tests that don't need to recover at all

Reply:
My guess is that most uses simply started by copying an existing test and then if they passed, probably changes were not considered. I'd suggest trying without recovery and then re-evaluate if the test fails.

Reply:
awesome, tested and the tests don't fail